cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -459,6 +459,69 @@ public String toString() {
   }
 
   /**
+   * Dataset Type Id identifies a given dataset module.
+   */
+  public static final class DatasetType {
+    private final Namespace namespace;
+    private final String typeId;
+
+    public DatasetType(final Namespace namespace, final String typeId) {
+      Preconditions.checkNotNull(namespace, "Namespace cannot be null.");
+      Preconditions.checkNotNull(typeId, "Dataset type id cannot be null.");
+      Preconditions.checkArgument(isValidDatasetId(typeId), "Invalid characters found in dataset type Id. '" +
+        typeId + "'. Module id can contain alphabets, numbers or _, -, . or $ characters");
+      this.namespace = namespace;
+      this.typeId = typeId;
+    }
+
+    public Namespace getNamespace() {
+      return namespace;
+    }
+
+    public String getNamespaceId() {
+      return namespace.getId();
+    }
+
+    public String getId() {
```

Comment:
This is confusing method name; for instance:
```
Id.DatasetType datasetTypeId = ...;
datasetTypeId.getId()
```

In other words, both the object is called Id and the method which returns a field is called getId.
Not sure if there's better name though; perhaps `getType`?